content/design/vendoring.md

```diff
@@ -53,3 +53,224 @@ already written):
+$ wc -l vendor/vendor.json
+181 vendor/vendor.json
+```
+
+This gives you assurance that the code generation libraries and
+templates will only change, for a given project, when you decide to
+explicitly do so, using your vendoring tool.
+
+Any future runs will deterministically generate the same code, on
+anyone's machine, unless you explicitly upgrade `goa`.
+
+
+## Generate your code
+
+Next step is to actually generate your code based on the design you
+wrote. Run your `gen/main.go` this way:
+
+```
+$ cd $GOPATH/src/github.com/your/pkg
+$ go run gen/main.go
+
+...
+swagger/swagger.json
+...
+app/contexts.go
+...
+tool/yourapi-cli/main.go
+tool/cli/commands.go
+...
+client/client.go
+...
+schema/schema.json
+...
+js/client.js
+```
+
+While you're at it, put that line near the top of `your/pkg/main.go`:
+
+```
+//go:generate go run gen/main.go
+```
+
+Next time, you only need to remember to run `go generate`. See
+[this article](https://blog.golang.org/generate) for more infos..
+
+
+## Vendoring dependencies of _generated_ code
+
+Once you've generated your code once, you'll see that there are more
+dependencies:
+
+```
+$ govendor list
+$ govendor list
+ v  github.com/dimfeld/httppath
+ v  github.com/goadesign/goa/design
+ v  github.com/goadesign/goa/design/apidsl
+ v  github.com/goadesign/goa/dslengine
+ v  github.com/goadesign/goa/goagen/codegen
+ v  github.com/goadesign/goa/goagen/gen_app
+ v  github.com/goadesign/goa/goagen/gen_client
+ v  github.com/goadesign/goa/goagen/gen_js
+ v  github.com/goadesign/goa/goagen/gen_main
+ v  github.com/goadesign/goa/goagen/gen_schema
+ v  github.com/goadesign/goa/goagen/gen_swagger
+ v  github.com/goadesign/goa/goagen/utils
+ v  github.com/goadesign/goa/version
+ v  github.com/manveru/faker
+ v  github.com/satori/go.uuid
+ v  github.com/zach-klippenstein/goregen
+ v  golang.org/x/tools/go/ast/astutil
+ v  gopkg.in/yaml.v2
+ e  github.com/armon/go-metrics
+ e  github.com/dimfeld/httppath
+ e  github.com/dimfeld/httptreemux
+ e  github.com/goadesign/goa
+ e  github.com/goadesign/goa/client
+ e  github.com/goadesign/goa/middleware
+ e  github.com/goadesign/goa/uuid
+ e  github.com/inconshreveable/mousetrap
+ e  github.com/satori/go.uuid
+ e  github.com/spf13/cobra
+ e  github.com/spf13/pflag
+ e  golang.org/x/net/context
+ e  golang.org/x/net/websocket
+pl  github.com/your/pkg
+ l  github.com/your/pkg/app
+ l  github.com/your/pkg/client
+ l  github.com/your/pkg/design
+pl  github.com/your/pkg/gen
+ l  github.com/your/pkg/js
+ l  github.com/your/pkg/tool/cli
+ ```
+
+See how we now have other external dependencies we need to pull in
+(marked `e` for `external`).  It includes the `goadesign/goa` package,
+which has most of the runtime components, to help you write your
+business logic.
+
+Vendor all of that in with:
+
+```
+$ govendor add +e
+$
+```
+
+And we're all good!  Now, not only code generation will be
+deterministically built, but also any reliance on goa, its depencies
+or your own dependencies.
```

Comment:
Not only are the code generation packages vendored but also `goa`, its dependencies and your dependencies.